File: 1512/solution.py
class Solution:
    def numIdenticalPairs(self, nums: list[int]) -> int:
        res = 0
        counts = {}
        for num in nums:
            if num in counts:
                res += counts[num]
                counts[num] += 1
            else:
                counts[num] = 1
        return res

        # time O(n)
        # space O(n)

File: 1512/solution.ts
function numIdenticalPairs(nums: number[]): number {
  const counts = new Map<number, number>();
  let res = 0;
  for (const num of nums) {
    res += counts.get(num) || 0;
    counts.set(num, (counts.get(num) || 0) + 1);
  }
  return res;
}
